src/webhook/validation/activegate.go

```diff
@@ -66,6 +81,65 @@ func missingActiveGateMemoryLimit(dv *dynakubeValidator, dynakube *dynatracev1be
 	return ""
 }
 
+func invalidActiveGateProxyUrl(dv *dynakubeValidator, dynakube *dynatracev1beta1.DynaKube) string {
+	if dynakube.Spec.Proxy != nil {
+		if len(dynakube.Spec.Proxy.ValueFrom) > 0 {
+			var proxySecret corev1.Secret
+			err := dv.clt.Get(context.TODO(), client.ObjectKey{Name: dynakube.Spec.Proxy.ValueFrom, Namespace: dynakube.Namespace}, &proxySecret)
+			if k8serrors.IsNotFound(err) {
+				return errorMissingActiveGateProxySecret
+			} else if err != nil {
+				return fmt.Sprintf("error occurred while reading PROXY secret indicated in the Dynakube specification (%s)", err.Error())
+			}
+			proxyUrl, ok := proxySecret.Data[agproxysecret.ProxySecretKey]
+			if !ok {
+				return errorInvalidProxySecretFormat
+			}
+			return validateProxyUrl(string(proxyUrl), errorInvalidProxySecretUrl, errorInvalidProxySecretEvalCharacter)
+		} else if len(dynakube.Spec.Proxy.Value) > 0 {
+			return validateProxyUrl(dynakube.Spec.Proxy.Value, errorInvalidActiveGateProxyUrl, errorInvalidEvalCharacter)
+		}
+	}
+	return ""
+}
+
 func memoryLimitSet(resources corev1.ResourceRequirements) bool {
 	return resources.Limits != nil && resources.Limits.Memory() != nil
 }
+
+// proxyUrl is valid if
+// 1) encoded
+// 2) "`\ are escaped using \
+func validateProxyUrl(proxyUrl string, parseErrorMessage string, evalErrorMessage string) string {
+	if parsedUrl, err := url.Parse(proxyUrl); err != nil {
+		return parseErrorMessage
+	} else {
+		password, _ := parsedUrl.User.Password()
+		if isEvalEscapeNeeded(password) {
+			return evalErrorMessage
+		}
+	}
+	return ""
+}
+
+// 'eval' command is used by entrypoint.sh:readSecret function to return its result.
+// For this reason quotation mark ", backtick ` and backslash \ characters have to be escaped using backslash.
+func isEvalEscapeNeeded(str string) bool {
```

Comment:
as far as I understand, this function would effectively be a block-list, blocking the three known characters. It would be more secure to restructure it to an allow-list, checking that every character is one of the known ones, i.e., letters, numbers and tested special characters and then checking if they are appropriately escaped

Reply:
my fault, characters can not be escaped because of dtclient. Finally, there are two invalid characters : apostrophe and backtick.